### PR TITLE
fix: union errors produce wrong message using

### LINKED ISSUE #12: Union errors produce wrong message using `.or()`
Error message not contains any info about wrong properties. But original zod error has all information.

Next code will print `Validation error: Invalid input at "responce-schema"`, but expected something like this `Validation error: Invalid literal value, expected "success" at "responce-schema.status"; Required at "responce-schema.data", or: Invalid literal value, expected "error" at "responce-schema.status"`

```ts
    const success = zod.object({
      status: zod.literal('success'),
      data: zod.object(),
    });

    const error = zod.object({
      status: zod.literal('error'),
    });

    const objSchema = success.or(error);

    try {
      objSchema.parse(
        {},
        {
          path: ['responce-schema'],
        }
      );
    } catch (err) {
      if (err instanceof ZodError) {
        console.log(fromZodError(err));
      }
    }
```

### ZodError
```
ZodError: [
  {
    "code": "invalid_union",
    "unionErrors": [
      {
        "issues": [
          {
            "code": "invalid_literal",
            "expected": "success",
            "path": [
              "responce-schema",
              "status"
            ],
            "message": "Invalid literal value, expected \"success\""
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "responce-schema",
              "message"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "code": "invalid_literal",
            "expected": "error",
            "path": [
              "responce-schema",
              "status"
            ],
            "message": "Invalid literal value, expected \"error\""
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "responce-schema",
              "message"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      }
    ],
    "path": [
      "responce-schema"
    ],
    "message": "Invalid input"
  }
]
```

### PR DESCRIPTION
**Pull request by Profesor08**
Closes #12

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "prettier/prettier": ["error", {}, { "usePrettierrc": true }],
     "simple-import-sort/imports": "error",
     "@typescript-eslint/ban-ts-ignore": "off",
-    "@typescript-eslint/ban-ts-comment": "off"
+    "@typescript-eslint/ban-ts-comment": "off",
+    "no-console": ["warn"]
```
Comment (jmike) [inline .eslintrc:17]: praise: great addition - thank you!

File: lib/ValidationError.spec.ts [hunk 2]
```diff
@@ -194,6 +194,152 @@ describe('fromZodError()', () => {
...
+                        "nestedObj",
+                        "name",
+                      ],
+                      "type": "string",
+                    },
+                  ]
+              `);
+      }
+    }
+  });
```
Comment (jmike) [inline lib/ValidationError.spec.ts:261]: issue: this test seems to be redundant as it is simply duplicating the code from above. What am I missing here?

File: lib/ValidationError.spec.ts [hunk 3]
```diff
@@ -194,6 +194,152 @@ describe('fromZodError()', () => {
...
+    });
+
+    const objSchema = success.or(error);
+
+    try {
+      objSchema.parse(
+        {},
+        {
+          path: ['responce-schema'],
+        }
```
Comment (jmike) [inline lib/ValidationError.spec.ts:282]: thought: unsure whether we should be passing `ParseParams` as part of this test. It seems to be a separate test case.

File: lib/ValidationError.ts [hunk 4]
```diff
@@ -18,18 +18,42 @@ export class ValidationError extends Error {
...
     issueSeparator?: string;
+    unionSeparator?: string;
     prefixSeparator?: string;
     prefix?: string;
   } = {}
 ): ValidationError {
   const {
     maxIssuesInMessage = 99, // I've got 99 problems but the b$tch ain't one
     issueSeparator = '; ',
+    unionSeparator = ', or: ',
```
Comment (jmike) [inline lib/ValidationError.ts:56]: issue: what if the user makes use of `zod.and()`. Wouldn't this return `or` by default? That doesn't seem right.

**Review by jmike — COMMENTED** [review]
Thank you for your PR - there's a couple of issues that require attention

File: lib/ValidationError.ts (hunk 4 above)
Comment (jmike) [reply lib/ValidationError.ts:56]: ignore the above comment. I just checked that in the case of `zod.and()` the output is NOT a union,

Comment (jmike) [inline lib/ValidationError.ts:56]: suggestion(nit): drop the `:` suffix and go for `, or ` as default value

File: lib/ValidationError.spec.ts (hunk 3 above)
Comment (Profesor08) [reply lib/ValidationError.spec.ts]: moved to separate test

File: lib/ValidationError.spec.ts (hunk 2 above)
Comment (Profesor08) [reply lib/ValidationError.spec.ts]: yes, didn't saw it

File: lib/ValidationError.ts (hunk 4 above)
Comment (Profesor08) [reply lib/ValidationError.ts]: good

**Review by jmike — APPROVED** [review]
💯 Awesome, let's do this

**Comment by mikemclin** [comment]
Looks like this was approved last week.  Excited for this to get deployed!  Thanks everyone :)

**Comment by jmike** [comment]
@Profesor08 thank you for your contribution - this PR is now merged. Version `0.3.1` has been released on npm.

**Pull request merged**